python/tank/platform/bundle.py

```diff
@@ -940,7 +959,21 @@ def __resolve_hook_path(self, settings_name, hook_expression):
             hooks_folder = os.path.join(engine.disk_location, "hooks")
             path = hook_expression.replace("{engine}", hooks_folder)
             path = path.replace("/", os.path.sep)
-        
+
+        elif hook_expression.startswith("{show}"):
+            # this is expected to be used only when there are show overrides, and hence
+            # possible declaration of correct hook name in the sgtk_config preferences files.
+            path = hook_expression.replace("{show}/", "")
```

Comment:
`path = hook_expression.replace("{show}", "SHARED/etc/sgtk/hooks")`

`"etc/sgtk/hooks"` is deprecated and shouldn't be used.

Reply:
Then I will need to add user_subdirectories and show_subdirectories differently.

Reply:
See the requested change in `expand_show_tree()` to include "SHARED" as one of the default branch_paths. In which case, this should just be:

```
path = hook_expression.replace("{show}", "etc/sgtk/hooks")
```
And `expand_show_tree()` will handle searching in SHARED for you.

Reply:
I will try and mimic the functionality of `preferences.Preferences.__init__()` for consistency. 
It has a bunch of search paths, that vary across facility("etc")/show("etc:SHARED/etc")/test("etc")/extra_subdirs("").